pkg/operator/ceph/pool/status.go

```diff
@@ -130,3 +131,27 @@ func toCustomResourceStatus(currentStatus *cephv1.MirroringStatusSpec, mirroring
 
 	return mirroringStatusSpec, mirroringInfoSpec, snapshotScheduleStatusSpec
 }
+
+func buildStatusInfo(cephBlockPool *cephv1.CephBlockPool) map[string]string {
+	m := make(map[string]string)
+	if cephBlockPool.Spec.Mirroring.Enabled {
+		mirroringInfo := opcontroller.GenerateStatusInfo(cephBlockPool)
```

Comment:
The mirroring info is only expected to be included if the pool was successfully configured. The `buildStatusInfo()` looks like it is being called from multiple places including just after initialization or failed reconciles, so we need to not include the mirroring status in that case. This would affect other dependencies outside rook that watch to know that the mirroring status indicates it was successfully configured.

Reply:
Good catch, I will fix this and include a test for it!